### ISSUE TITLE
Connexion impossible au site GRDF

### ISSUE THREAD

**Issue by didierhassio** [yukulehe/gazpar2mqtt#78]
Bonjour,
Je n'arrive pas à me connecter au site GRDF, je pense que cela vient du capcha...
par ailleurs, je me connecte sans problème via mon navigateur internet.
Merci d'avance,

`2023-01-20 11:53:58,110 INFO #                Program parameters                       #
2023-01-20 11:53:58,111 INFO -----------------------------------------------------------
2023-01-20 11:53:58,111 INFO GRDF config : username = ******@****.**, password = ******
2023-01-20 11:53:58,111 INFO MQTT broker config : host = core-mosquitto, port = 1883, clientId = gazpar2mqtt, qos = 0, topic = gazpar2mqtt, retain = True, ssl = False
2023-01-20 11:53:58,111 INFO Standlone mode : Enable = True
2023-01-20 11:53:58,111 INFO Home Assistant discovery : Enable = False
2023-01-20 11:53:58,112 INFO Thresold options : Warning percentage = 80
2023-01-20 11:53:58,112 INFO Influxdb config : Enable = False
2023-01-20 11:53:58,112 INFO Price config : Default price = 0.07 €/kWh, default fix price = 0.9 €/day, path to file = /data
2023-01-20 11:53:58,112 INFO Database options : Force reinitialization = False, Path = /data
2023-01-20 11:53:58,112 INFO Debug mode : Enable = False
2023-01-20 11:53:58,113 INFO Check parameters...
2023-01-20 11:53:58,113 INFO Parameters are ok !
2023-01-20 11:53:58,113 INFO -----------------------------------------------------------
2023-01-20 11:53:58,113 INFO #        Connexion to SQLite database                     #
2023-01-20 11:53:58,113 INFO -----------------------------------------------------------
2023-01-20 11:53:58,113 INFO Connexion to SQLite database...
2023-01-20 11:53:58,119 INFO SQLite database connected !
2023-01-20 11:53:58,121 INFO Last execution date 2023-01-20 10:49:33.767458, program was in version 0.8.2.
2023-01-20 11:53:58,121 INFO Checking database version...
2023-01-20 11:53:58,122 INFO Your database is already up to date : version 0.8.0.
2023-01-20 11:53:58,123 INFO Retrieve database statistics...
2023-01-20 11:53:58,124 INFO 0 informatives measures stored
2023-01-20 11:53:58,124 INFO 0 PCE(s)
2023-01-20 11:53:58,124 INFO First measure : None
2023-01-20 11:53:58,126 INFO Last measure : None
2023-01-20 11:53:58,126 INFO -----------------------------------------------------------
2023-01-20 11:53:58,127 INFO #              Connexion to Mqtt broker                   #
2023-01-20 11:53:58,128 INFO -----------------------------------------------------------
2023-01-20 11:53:58,128 INFO Connect to Mqtt broker...
2023-01-20 11:54:02,141 INFO Mqtt broker connected !
2023-01-20 11:54:02,142 INFO -----------------------------------------------------------
2023-01-20 11:54:02,142 INFO #            Get data from GRDF website                   #
2023-01-20 11:54:02,142 INFO -----------------------------------------------------------
2023-01-20 11:54:02,143 INFO Connexion to GRDF, try 1/14...
2023-01-20 11:54:02,450 INFO <Response [200]>
2023-01-20 11:54:02,451 INFO <RequestsCookieJar[<Cookie XSRF-TOKEN=ab53517c-8e29-4361-9047-85e20133110e for .grdf.fr/>, <Cookie ROUTEID=.1 for login.monespace.grdf.fr/>, <Cookie JSESSIONID=74BA23CFAE3258B0A86168E1DC709938 for login.monespace.grdf.fr/sofit-account-api>, <Cookie BIGipServer~EXT_PROD~PL-EXT_PROD-VS_P1_SIG-S_EPS_HTTPS_Internet=rd6o00000000000000000000ffff0a28a21ao443 for monespace.grdf.fr/>, <Cookie auth_nonce=dd9fc31e15d70d4a0bc4ee5c56055f5a for monespace.grdf.fr/>, <Cookie auth_redir=https://monespace.grdf.fr/ for monespace.grdf.fr/>]>
2023-01-20 11:54:02,452 INFO Login unsuccessful. Invalid returned information: {"state":"REINPUT","error":"LOGIN_INVALID_ATTEMPTS","actualLockoutDuration":0,"message":"login.error.invalidattempts","displayCaptcha":true,"surname":"SURNAME","name":"NAME","email":"xxxxxxxxxx@outlook.fr"}
2023-01-20 11:54:02,452 INFO Unable to login to GRDF website
2023-01-20 11:54:02,453 INFO Wait 20 seconds (0 min) before next try`

**Comment by goldendatahub** [comment]
Hello. Est-ce que ton mot de passe contient des caractères spéciaux ? Si c'est le cas, il faut que tu changes ton mot de passe.

**Comment by didierhassio** [comment]
J'avais de caractères spéciaux et je les ai remplacé, mais cela ne change rien.

**Comment by didierhassio** [comment]
lorsque je me connecte via mon navigateur internet, je dois valider un captcha sinon la connexion est refusée.

**Comment by goldendatahub** [comment]
Yes, mais ça fait un moment que le Captcha existe, et pourtant il ne posait pas trop de problème jusque là pour le script.

**Comment by didierhassio** [comment]
J'ai vu que cette intégration avait un problème avec le captcha, c'est pour cela que je pense que cela vient de là.
[https://github.com/frtz13/homeassistant_gazpar_cl_sensor](url)

**Comment by goldendatahub** [comment]
Tu as sans doute raison. Le captcha est l'épée de Damoclès. On sait tous qu'un jour on allait être bloqué. Ce jour vient peut être d'arriver.

**Comment by airvb** [comment]
Tout pareil ici ... bad news...

**Comment by chris60600** [comment]
Je confirme, le Captcha est maintenant systématique pour se connecter. Dommage

**Comment by JigSawFr** [comment]
Dommage, idem, le captcha est systématique.

Peut être voir pour implémenter des solveurs comme https://2captcha.com/, des librairies python sont dispo: https://github.com/2captcha/2captcha-python

**Comment by airvb** [comment]
> Peut être voir pour implémenter des solveurs comme https://2captcha.com/, des librairies python sont dispo: https://github.com/2captcha/2captcha-python

Le tarif reste raisonnable, cela peut il fonctionner pour ce type de captcha ?

**Comment by JigSawFr** [comment]
Yes il faut juste l'implémenter ici

**Comment by dudu7731** [comment]
Bonjour,

dans le même style j'ai ça de mon coté : 
Login unsuccessful. Invalid returned information: {"state":"REINPUT","error":"**CAPTCHA_FAIL**","actualLockoutDuration":0,"message":"login.error.recaptchako","displayCaptcha":true,"surname":"XXXXX","name":"XXXXX","email":"XXX@gmail.com"}

Le CAPTCHA bloque bien ....

**Comment by mdeweerd** [comment]
Pour info, une alternative est en cours de mise au point.
Pour les évolutions suivre [fil de discussion sur hacf.fr](https://forum.hacf.fr/t/gazpar-grdf-ko-suite-exigence-captcha/20004?u=le_top).
Il n'y a pas de connecteur MQTT dedans actuellement.

**Comment by airvb** [comment]
En attendant ...

Un dirty script pour descendre les données de GRDF vers influx 2.0 avec un dashboad grafana pour visualiser tout ça !
Pas de faux espoir il faut télécharger le fichier .xlsx sur monespace.grdf.fr

https://github.com/airvb/grdf2influx2

![This is an image](https://nc.airvb.com/index.php/s/bzs9nqAt6YqXYap/preview)
